command/validate/validate.go

```diff
@@ -60,6 +60,9 @@ func (c *cmd) Run(args []string) int {
 		c.UI.Error(fmt.Sprintf("Config validation failed: %v", err.Error()))
 		return 1
 	}
+	for _, w := range b.Warnings {
+		c.UI.Warn(w)
+	}
```

Comment:
I wonder if this should be inside the `!c.quiet` block.

Reply:
Good question, although warnings do not constitute an error, they are still a potential misconfiguration. 🤔 

Perhaps the real question is: are warnings an error or not? If they are, they should cause a failure. If they are not, then they should be in the quiet block.

As a user, I would probably want `consul validate` to error on any warnings as well, since I am actively attempting to check my configuration. Changing those semantics now might be annoying to current users though, but perhaps helpful.

Reply:
A new flag that exits non-zero if there are warnings would be a good idea. I'll add that as a follow up.

For the current behaviour it sounds like you agree this should only be printed when not `--quiet`, so I will make that change.

Reply:
That makes sense to me. 👍